packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -234,98 +251,121 @@ const RelationModalRenderer = ({
-                label={formatMessage({
-                  id: 'content-manager.components.RelationInputModal.button-fullpage',
-                  defaultMessage: 'Go to entry',
-                })}
+                marginRight={1}
               >
-                <ArrowsOut />
+                <ArrowLeft />
               </IconButton>
+              <Typography tag="span" fontWeight={600}>
+                {isCreating
+                  ? formatMessage({
+                      id: 'content-manager.relation.create',
+                      defaultMessage: 'Create a relation',
+                    })
+                  : formatMessage({
+                      id: 'content-manager.components.RelationInputModal.modal-title',
+                      defaultMessage: 'Edit a relation',
+                    })}
+              </Typography>
             </Flex>
-          </Modal.Header>
-          <Modal.Body>
-            <FormContext
-              method="PUT"
-              initialValues={currentDocument.getInitialFormValues()}
-              validate={(values: Record<string, unknown>, options: Record<string, string>) => {
-                const yupSchema = createYupSchema(
-                  currentDocument.schema?.attributes,
-                  currentDocument.components,
-                  {
-                    status: currentDocument.document?.status,
-                    ...options,
-                  }
-                );
-
-                return yupSchema.validate(values, { abortEarly: false });
+            <IconButton
+              onClick={() => {
+                dispatch({
+                  type: 'GO_FULL_PAGE',
+                });
+                if (!state.hasUnsavedChanges) {
+                  navigate(getFullPageUrl(currentDocumentMeta));
+                }
               }}
+              variant="tertiary"
+              label={formatMessage({
+                id: 'content-manager.components.RelationInputModal.button-fullpage',
+                defaultMessage: 'Go to entry',
+              })}
             >
-              {children}
-            </FormContext>
-          </Modal.Body>
-        </StyledModalContent>
-      </Modal.Root>
-    </RelationModalProvider>
+              <ArrowsOut />
+            </IconButton>
+          </Flex>
+        </Modal.Header>
+        <Modal.Body>
+          <FormContext
+            method="PUT"
+            initialValues={currentDocument.getInitialFormValues()}
+            validate={(values: Record<string, unknown>, options: Record<string, string>) => {
+              const yupSchema = createYupSchema(
+                currentDocument.schema?.attributes,
+                currentDocument.components,
+                {
+                  status: currentDocument.document?.status,
+                  ...options,
+                }
+              );
+
+              return yupSchema.validate(values, { abortEarly: false });
+            }}
+          >
+            <RelationModalBody />
```

Comment:
@remidej you warned me about doing this but I did it anyway 😬 . I haven't had any problems but wanted to point it out in case you know exactly case where this might introduce a bug.